spec/components/url_helper_spec.rb

```diff
@@ -139,6 +139,22 @@
     end
   end
 
+  describe "#rails_route_from_url" do
+    it "recognizes a user path" do
+      result = UrlHelper.rails_route_from_url('http://example.com/u/john_smith')
+      expect(result[:controller]).to eq("users")
+      expect(result[:action]).to eq("show")
+      expect(result[:username]).to eq("john_smith")
+    end
+
+    it "recognizes a user path with unicode characters in the username" do
+      result = UrlHelper.rails_route_from_url('http://example.com/u/björn_ulvaeus')
+      expect(result[:controller]).to eq("users")
+      expect(result[:action]).to eq("show")
+      expect(result[:username].force_encoding('UTF-8')).to eq("björn_ulvaeus")
```

Comment:
I'm not sure how I feel about this force_encoding. Is there a better way to do it?

Reply:
I think it's probably ok